src/helpers/helpers.options.js

```diff
@@ -175,3 +175,64 @@ export function resolve(inputs, context, index, info) {
 		}
 	}
 }
+
+const notIndexable = ['borderDash', 'fill', 'events'];
+const scriptable = (value) => typeof value === 'function';
+const indexable = (key, value) => isArray(value) && notIndexable.indexOf(key) === -1;
+const readKey = (prefix, name) => prefix ? prefix + _capitalize(name) : name;
+const defined = (value) => typeof value !== 'undefined';
+
+/**
+ * Creates a Proxy for resolving raw values for options.
+ * @param {object[]} scopes - The option scopes to look for values, in resolution order
+ * @param {string[]} [prefixes] - The prefixes for values, in resolution order.
+ * @private
+ */
+export function _resolver(scopes, prefixes = ['']) {
+	const cache = Object.create(null);
+	return new Proxy(cache, {
+		get(target, prop) {
+			let value = Reflect.get(target, prop);
+			if (defined(value)) {
+				return value;
+			}
+			let key;
+			loop:
```

Comment:
I'm not sure if labels + break are bad style or not, but it does simplify this loop.

Reply:
Yeah, I hate to write same checks to two levels of loop. Maybe this should be split into sub functions, but would like to see the performance first, before tuning for maintainability.